connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -890,7 +891,49 @@ object SparkSession extends Logging {
     // the remote() function, as it takes precedence over the SPARK_REMOTE environment variable.
     private val builder = SparkConnectClient.builder().loadFromEnvironment()
     private var client: SparkConnectClient = _
-    private[this] val options = new scala.collection.mutable.HashMap[String, String]
+    private var server: Option[Process] = None
+    private[this] lazy val options = {
+      val configs: Map[String, String] =
+        sys.props.filter(p => p._1.startsWith("spark.sql.") && p._2.nonEmpty).toMap
+      val map = new scala.collection.mutable.HashMap[String, String]
+      map.addAll(configs)
+    }
+    private def remoteString: Option[String] = {
+      options
+        .get("spark.remote")
+        .orElse(Option(System.getProperty("spark.remote"))) // Set from Spark Submit
+        .orElse(sys.env.get(SparkConnectClient.SPARK_REMOTE))
+    }
+
+    private def withLocalConnectServer[T](f: => T): T = {
```

Comment:
Is this normal? I though the submit would take care of this for us?

Reply:
It will start a Spark Connect server that starts the local cluster in this case. And then it returns a client to it.